core/src/datayoga_core/blocks/relational/write/block.py

```diff
@@ -243,3 +217,20 @@ def execute_delete(self, records: List[Dict[str, Any]]):
 
     def stop(self):
         self.dispose_engine()
+
+    def _is_connection_valid(self) -> bool:
+        """Checks if the current database connection is still valid."""
+        try:
+            # Execute a simple query to check if the connection is still valid
+            self.connection.scalar(select(1))
```

Comment:
Why not use the built in method? This may not work for Oracle that needs Dual

Reply:
changed implementation.